alert/interface.go

```diff
@@ -0,0 +1,48 @@
+package alert
+
+import "time"
+
+const (
+	NoAlertFound Error = "no alert found for the input key"
+)
+
+// Error implements Error interface
+type Error string
+
+// Error reports underlying error as a string
+func (e Error) Error() string {
+	return string(e)
+}
+
+// Altertable defines an interface that can be managed using Alerter interface.
+// An instance of Alertable is an alert object
+type Alertable interface {
```

Comment:
Isn't this interface already implemented in api.pb.go as the Alert struct is a proto definition?

Reply:
considering this is going in libopenstorage, it would be easier to base it on interface instead of concrete structs. Other (non px) implementors may want to do have different definitions in their struct but as long as they comply with interface, we should be good... so was my thinking.